### PR TITLE
bugfix: resolve issue in Raft model a follower's crash may lead to the continued use of expired tokens

### LINKED ISSUE #6917: In the Raft model a follower's crash may lead to the continued use of expired tokens.
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description

In the Raft model a follower's crash may lead to the continued use of expired tokens.

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. raft模式下，下线一台follower
2. 恰好此时client侧token过期，进行刷新
3. 恰好刷新的请求到了离线的follower中，导致请求失败
4. 由于请求之前将时间戳进行了更新，导致过期的token被视为可用

a. In the Raft model, a follower goes offline.
b. At this moment, the token on the client side expires and needs to be refreshed.
c. The refresh request happens to reach the offline follower, resulting in a failed request.
d. Because the timestamp was updated prior to the request, the expired token is mistakenly considered valid.

Minimal yet complete reproducer code (or URL to code):
RaftRegistryServiceImpl#refreshToken中先更新了tokenTimeStamp，才发出获取token的请求，而且并没有判断httpstatus是否正常。
In RaftRegistryServiceImpl#refreshToken, the tokenTimeStamp is updated before the request to obtain the token is sent, and there is no check for whether the HTTP status is normal.
```
    private static void refreshToken(String tcAddress) throws RetryableException {
        // if username and password is not in config , return
        if (StringUtils.isBlank(USERNAME) || StringUtils.isBlank(PASSWORD)) {
            return;
        }
        // get token and set it in cache
        Map<String, String> param = new HashMap<>();
        param.put(PRO_USERNAME_KEY, USERNAME);
        param.put(PRO_PASSWORD_KEY, PASSWORD);
        Map<String, String> header = new HashMap<>();
        header.put(HTTP.CONTENT_TYPE, ContentType.APPLICATION_JSON.getMimeType());
        String response = null;
        tokenTimeStamp = System.currentTimeMillis();
        try (CloseableHttpResponse httpResponse =
            HttpClientUtil.doPost("http://" + tcAddress + "/api/v1/auth/login", param, header, 1000)) {
            if (httpResponse != null) {
                if (httpResponse.getStatusLine().getStatusCode() == HttpStatus.SC_OK) {
                    response = EntityUtils.toString(httpResponse.getEntity(), StandardCharsets.UTF_8);
                    JsonNode jsonNode = OBJECT_MAPPER.readTree(response);
                    String codeStatus = jsonNode.get("code").asText();
                    if (!StringUtils.equals(codeStatus, "200")) {
                        //authorized failed,throw exception to kill process
                        throw new AuthenticationFailedException("Authentication failed! you should configure the correct username and password.");
                    }
                    jwtToken = jsonNode.get("data").asText();
                } else {
                    //authorized failed,throw exception to kill process
                    throw new AuthenticationFailedException("Authentication failed! you should configure the correct username and password.");
                }
            }
        } catch (IOException e) {
            throw new RetryableException(e.getMessage(), e);
        }
    }
```

### Ⅵ. Environment:

- JDK version(e.g. `java -version`):
- Seata client/server version:  
- Database version:
- OS(e.g. `uname -a`):
- Others:

### PR DESCRIPTION
**Pull request by lixingjia77**
… client to continued use of expired tokens

- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did

解决raft模式下，follower崩溃可能导致client继续使用过期令牌的问题
resolve issue in Raft model where follower crashes could lead client to continued use of expired tokens

### Ⅱ. Does this pull request fix one issue?

fixes #6917

In RaftRegistryServiceImpl#refreshToken, the tokenTimeStamp is updated before the request to obtain the token is sent, and there is no check for whether the HTTP status is normal.

### CODE REVIEW

File: changes/zh-cn/2.x.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 - [[#6899](https://github.com/apache/incubator-seata/pull/6899)] 修复file.conf打包后的读取
 - [[#6890](https://github.com/apache/incubator-seata/pull/6890)] 修复saga设计json转标准json过程中: 子状态机补偿节点无法被识别
 - [[#6907](https://github.com/apache/incubator-seata/pull/6907)] 修复Codecov未生成报告的问题
+- [[#6925](https://github.com/apache/incubator-seata/pull/6925)] 修复Raft模式下，Follower崩溃可能导致Client继续使用过期令牌的问题
```
Comment (funky-eyes) [inline changes/zh-cn/2.x.md:15]: 最下方的开发者信息也需要登记
The developer information at the bottom also requires registration.

File: changes/en-us/2.x.md [hunk 2]
```diff
@@ -12,6 +12,7 @@ Add changes here for all PR submitted to the 2.x branch.
 - [[#6899](https://github.com/apache/incubator-seata/pull/6899)] fix file.conf read failed after package
 - [[#6890](https://github.com/apache/incubator-seata/pull/6890)] fix designerJson to standardJson: subStateMachine compensateState cannot be recognized
 - [[#6907](https://github.com/apache/incubator-seata/pull/6907)] fix the issue of Codecov not generating reports
+- [[#6925](https://github.com/apache/incubator-seata/pull/6925)] fix the issue in Raft model a follower's crash may lead to the continued use of expired tokens
```
Comment (funky-eyes) [inline changes/en-us/2.x.md:15]: 最下方的开发者信息也需要登记
The developer information at the bottom also requires registration.

Comment (lixingjia77) [reply changes/en-us/2.x.md:15]: Thanks for your reminder, the developer info has been added.

**Review by funky-eyes — APPROVED** [review]
LGTM

**Pull request merged**